File: rust/src/generic/i_exponent.rs
pub trait IExponent {
    fn add(&self, o: &Self) -> Self;
    fn sub(&self, o: &Self) -> Self;
    fn lcm(&self, other: &Self) -> Self;
    fn degree(&self) -> u32;
    fn lex_compare(&self, other: &Self) -> std::cmp::Ordering;
    fn can_reduce(&self, divisor: &Self) -> bool;

}

File: typescript/generic/iExponent.ts
export interface IExponent<T> {
  add(o: T): T;
  sub(o: T): T;
  lcm(o: T): T;
  deg(): number;
  lexCompare(o: T): number;
  canReduce(o: T): boolean;
  equals(o: T): boolean;
}